Sources/Datadog/Core/Feature.swift

```diff
@@ -143,8 +143,8 @@ internal struct FeatureUpload {
     init(
         featureName: String,
         storage: FeatureStorage,
-        uploadHTTPHeaders: HTTPHeaders,
-        uploadURLProvider: UploadURLProvider,
+        uploadHTTPHeadersProvider: HTTPHeadersProvider,
+        uploadURL: UploadURL,
```

Comment:
i think these 2 are not useful individually and they always go together.
so we can merge them into one single `UploadRequestProvider` component?

rough sketch of `UploadRequestProvider`:
```swift
class UploadRequestProvider {
  let baseURL: URL
  let queries: [Query]
  let headers: [HTTPHeader]
  ...
  func buildRequest() -> URLRequest { ... }
}
```
that way we can pass one less parameter to `Feature` init methods and have one single component who is responsible for creating `URLRequest`.

as far as i see, currently we pass `HTTPHeadersProvider` and `uploadURL` separately down until `DataUploader` and then it creates `URLRequest` by using both. am i missing a point?

Reply:
Yes, totally - merging both makes sense now as the `URL` logic is pretty straightforward. Done in https://github.com/DataDog/dd-sdk-ios/pull/562/commits/3b0f69032a817ab4e0369e2fe29f12db2af318a8 with introduction of `RequestBuilder`.